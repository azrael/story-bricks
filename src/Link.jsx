import React from 'react';
import pt from 'prop-types';
import s from './story.css';

const Link = ({ href, blank, children }) => {
    if (!href) {
        return <span className={s.link}>
            {children}
        </span>;
    }

    return <a
        className={s.link}
        href={href}
        target={blank ? '_blank' : '_self'}
        rel='noreferrer noopener'
    >
        {children}
    </a>;
};

Link.displayName = 'Link';

Link.propTypes = {
    href: pt.string,
    blank: pt.bool
};

Link.defaultProps = {
    blank: false
};

export default Link;
