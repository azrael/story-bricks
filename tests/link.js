import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../src/Link.jsx';

it('Link with href renders correctly', () => {
    const tree = renderer
        .create(<Link href='/foo'>test</Link>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('Link without href renders correctly', () => {
    const tree = renderer
        .create(<Link>test</Link>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('Link with blank renders correctly', () => {
    const tree = renderer
        .create(<Link href='/foo' blank>test</Link>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('Link with blank but without href renders correctly', () => {
    const tree = renderer
        .create(<Link blank>test</Link>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
