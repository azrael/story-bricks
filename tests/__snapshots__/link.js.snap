// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link with blank but without href renders correctly 1`] = `
<span
  className="link"
>
  test
</span>
`;

exports[`Link with blank renders correctly 1`] = `
<a
  className="link"
  href="/foo"
  rel="noreferrer noopener"
  target="_blank"
>
  test
</a>
`;

exports[`Link with href renders correctly 1`] = `
<a
  className="link"
  href="/foo"
  rel="noreferrer noopener"
  target="_self"
>
  test
</a>
`;

exports[`Link without href renders correctly 1`] = `
<span
  className="link"
>
  test
</span>
`;
